app/models/get_an_identity/user.rb

```diff
@@ -0,0 +1,54 @@
+module GetAnIdentity
```

Comment:
Im not sure if this should be a model? feels like this should be a service as your fetching a user? Your not creating etc..

Reply:
Also confusing as this is called from the services folder right?

Reply:
Also, models don't need to create things. User.find doesn't create anything either.

Reply:
What you're saying there is how I'm viewing them. They're objects representing data from another DB, so they're models in the same way as our regular models they just have a bit of distance between them and their data store. What I mean by other models is what's in lib/ecf_api, they're models they're just not in the model directory.

I could move them under lib, but it feels odd to have a generic "lib" directory containing code, the ecf_api classes feel better suited to move over into models to fit them into the core Rails MVC structure. It's not really particularly important, one or the other they're accessed and represented the same sort of thing.

Reply:
I feel its a bit of a code smell to be honest, its obvious not our fault how the data model is setup and requiring to access this data in ECF. Although in a pure rails model youl would be able to create/delete/update all through the model class. I see this as a read only action in finding out the state of the data and although i actually dont agree it should sit in lib either.

i would rather we had something like `app/services/external/ecf/user_details.rb` or `..find_user.rb` / `retrieve_user.rb` Ideally we would remove the `app/lib` folder completely as i think all those folders could be in better places. By having services like this that have one task which is getting the latest details from ecf app keeps it very separate and unlikely cause future confusion. 

Also this is all for data-syncing between apps right? i do worry how much this could grow in future but so far we have limited models in NPQ so i doubt will become massive concern. Possibly as just us two happy to have someone else weigh in on the proposal?

Reply:
I can see that as an alternative, I disagree it's a code smell and would argue that pushing more into services isn't a great move since services are themselves non-standard Rails and turn into a catch-all "things that don't fit neatly into MVC." This does, to me, fit neatly into MVC as it fits pretty cleanly as a data-model.

@peteryates I think Sam has asked for a third opinion on this one.

Reply:
This is an interesting one and there are pros and cons to both approaches. I'm not against the idea of API-backed models but as @shjohnson says, it can lead to confusion when the model doesn't feel/act quite like an ActiveModel. 

The conventional way to do this in Rails would be via [ActiveResource](https://github.com/rails/activeresource), which does give you a lot of the Railsisms for 'free'. Perhaps that pattern could be applied here? That would make it more transparent and resemble the sibling classes better - providing the remote API will play nicely with it.

ActiveResource supports creating/updating/deleting too, and from what I can see this class just retrieves and makes available data - so perhaps a service class makes more sense.

Regardless, I'd probably go with either:

*  if it's only ever going to retrieve one user record at once change it to a service class `GetAnIdentityAPI::RetrieveUser` - so it's really obvious what it's doing
* leave it as a model but namespace it so it's clear it uses an API and not the local db

Reply:
I definitely like the idea of ActiveResource, though frankly having used it in the past I'm wary of getting stuck in with it again as it ended up being very slow compared to alternatives. This may have changed though.

With this one running over quite a bit I'm not keen to refactor into a `GetAnIdentityAPI::RetrieveUser` style service, though I can see that approach working. It doesn't quite match the pattern we have with the EcfApi namespaced models though.

With this already being namedspaced under GetAnIdentity, which should make it clear it's representing GetAnIdentity models, do you think that matches what you mean by the second point? Or do you think a specific `External` namespace?

Reply:
`External` works for me, it makes it clear there's a distinction between it and the other models. @shjohnson?

Reply:
I've pushed up a commit that puts them under `::GetAnIdentity::API`, so `::GetAnIdentity::API::AccessToken` and `::GetAnIdentity::API::User`, this feels the most descriptive and fits in with the naming pattern of `EcfApi::User` etc.

Reply:
Sorry i never responded to this but discussed with @Lockyy on Thursday. I agree to make things super clear about it being separate and external works fine. Honestly i would be cautious about this pattern and if this is the only task that stays in that folder i feel a top level service folder would be better than being placed in `app/lib`.

Then following that we could move the other items in lib into better folder structures and get rid of `app/lib`?